cmd/network/verification.go

```diff
@@ -0,0 +1,114 @@
+	}
+	return regionDefault
+
+}
+func NewCmdValidateEgress() *cobra.Command {
+	config := egressConfig{}
+
+	validateEgressCmd := &cobra.Command{
+		Use:   "verify-egress",
+		Short: "Verify essential openshift domains are reachable from given subnet ID.",
+		Long:  `Verify essential openshift domains are reachable from given subnet ID.`,
+		Example: `For AWS, ensure your credential environment vars 
+AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY (also AWS_SESSION_TOKEN for STS credentials) 
+are set correctly before execution.
+
+# Verify that essential openshift domains are reachable from a given SUBNET_ID
+osdctl network verify-egress --subnet-id $(SUBNET_ID) --image-id $(IMAGE_ID)`,
+		Run: func(cmd *cobra.Command, args []string) {
+			runEgressTest(config)
+			},
+	}
+
+	validateEgressCmd.Flags().StringVar(&config.vpcSubnetID, "subnet-id", "", "source subnet ID")
+	validateEgressCmd.Flags().StringVar(&config.cloudImageID, "image-id", "", "(optional) cloud image for the compute instance")
+	validateEgressCmd.Flags().StringVar(&config.instanceType, "instance-type", "t3.micro", "(optional) compute instance type")
+	validateEgressCmd.Flags().StringVar(&config.region, "region", getDefaultRegion(), fmt.Sprintf("(optional) compute instance region. If absent, environment var %[1]v will be used", regionEnvVarStr, regionDefault))
+	validateEgressCmd.Flags().StringVar(&config.region, "profile", "", fmt.Sprintf("(optional) AWS profile. If present, any credentials passed with CLI will be ignored."))
+	validateEgressCmd.Flags().StringToStringVar(&config.cloudTags, "cloud-tags", defaultTags, "(optional) comma-seperated list of tags to assign to cloud resources e.g. --cloud-tags key1=value1,key2=value2")
+	validateEgressCmd.Flags().BoolVar(&config.debug, "debug", false, "(optional) if true, enable additional debug-level logging")
+	validateEgressCmd.Flags().DurationVar(&config.timeout, "timeout", 1*time.Second, "(optional) timeout for individual egress verification requests")
+	validateEgressCmd.Flags().StringVar(&config.kmsKeyID, "kms-key-id", "", "(optional) ID of KMS key used to encrypt root volumes of compute instances. Defaults to cloud account default key")
+
+	if err := validateEgressCmd.MarkFlagRequired("subnet-id"); err != nil {
+		validateEgressCmd.PrintErr(err)
+		os.Exit(1)
+	}
+
+	return validateEgressCmd
+
+}
+
+func runEgressTest(config egressConfig) {
+
+			// ctx
+			ctx := context.TODO()
+
+			// Create logger
+			builder := ocmlog.NewStdLoggerBuilder()
+			builder.Debug(config.debug)
+			logger, err := builder.Build()
+			if err != nil {
+				fmt.Printf("Unable to build logger: %s\n", err.Error())
+				os.Exit(1)
+			}
+
+			logger.Info(ctx, "Using region: %s", config.region)
+			var creds interface{}
+			if config.awsProfile != "" {
+				creds = config.awsProfile
+			} else {
+				creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+			}
```

Comment:
```suggestion
			creds = config.awsProfile {
		        if creds == "" {
				creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
			}
```

Reply:
I actually prefer the first way.

Reply:
No worries, it was just a suggestion, not a request for changes.